docs/development/decision-records/2024-09-16-proposal-federated-catalog-distribution/README.md

```diff
@@ -0,0 +1,88 @@
+# Proposal for FederatedCatalog with Tractus-X distribution and its TargetNodeDirectory
+
+## Decision
+
+Federated Catalog will be deployable as a standalone component capable of crawling all the chosen catalogs and expose that data. The Tractus-Connector Helm charts will be updated to feature a new Federated Catalog deployment template.
+Regarding the TargetNodeDirectory, a new extension in the FederatedCatalog will have a db/cache containing the BPNL's and Connectors' URL's of each partner a member wants the offers from.
+
+## Rationale
+
+Considering the Federated Catalog distribution, choosing a solution that decouples it from the Control Plane (like the one used for the Data Plane) and able to be scalable will future-proof the Federated Catalog as a feature and embraces wider usage.
+Having a specific runtime incurs on additional overhead (new Helm Chart, as example) and results in additional configuration complexity. Also, periodical crawling results in increased remote calls over time. However, being decoupled from the Control Plane, this solution permits the Federated Catalog to scale independently (based on own demand).
+
+For TargetNodeDirectory it will be set by a new extension responsible for exposing an API, where a member can input the BPNL's of the participants from which the catalogs are wanted, and then it will retrieve and store the respective Connector URL's. This new extension would get the data from the Discovery Service, provided a BPNL, and will be named `DiscoveryServiceRetrieverExtension`. This solution allows the member to choose precisely the Target Catalog Nodes that interests them, resulting in reduced network calls and latency.
+Additionally, if a Connector URL is registered (or unregistered) in the Discovery Service, the retriever will reflect it since it requests based on BPNL (which should not change) and the registered URL's will be returned.
+
+This solution improves on the default one of having the data in a static file since a dynamic approach would avoid downtime when a change is required.
```

Comment:
This should go in the `Approach` section, as it outlines technical implementation. 
It is enough to say something to the effect of "target node data is sourced directly from the BPNL Directory service (or whatever it's called) since that already exists in Tractus-X and provides comprehensive participant data." It might also be worth mentioning, that there currently is no way for participants _not to be listed_ in that TND.

Apart from that, there is no need to explain the TND's functionality here. Rather, in the "Approach" section, please detail where the TND gets its data, if and how it's cached + what cache eviction strategy will be used, how new (or deleted) entries are detected etc.

Reply:
Why are we using BPNs when we are planning to move away from those at the protocol layer? We should be entering the DIDs. This means we need to define an easy way for people to find the DID, which can be done using the BDRS client.

Reply:
Thank you, I understand it and updated the documentation to include DID input, having the BDRS resolve of those DID and call the Discovery Service to obtain the Connector URL's. 
This is aligned to what was discussed in a weekly meeting and is updated [here](https://github.com/eclipse-tractusx/tractusx-edc/pull/1555/commits/4b75211ca9ffac01da973e2f29f110cb18fc3e76).